File: python/UtopianTree.py
#!/bin/python3

import math
import os
import random
import re
import sys

# Complete the utopianTree function below.
def utopianTree(n):
    result = 1
    for i in range(n):
        if i % 2 == 0:
            result *= 2
        else:
            result += 1
            
    return result

if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    t = int(input())

    for t_itr in range(t):
        n = int(input())

        result = utopianTree(n)

        fptr.write(str(result) + '\n')

    fptr.close()

File: scala/Utopian_Tree.scala
import java.io._
import java.math._
import java.security._
import java.text._
import java.util._
import java.util.concurrent._
import java.util.function._
import java.util.regex._
import java.util.stream._

object Solution {

    // Complete the utopianTree function below.
    def utopianTree(n: Int): Int = {
        var result = 1
        for (i <- 0 until n)
            if (i % 2 == 0)
                result *= 2
            else
                result += 1
                
        return result
    }

    def main(args: Array[String]) {
        val stdin = scala.io.StdIn

        val printWriter = new PrintWriter(sys.env("OUTPUT_PATH"))

        val t = stdin.readLine.trim.toInt

        for (tItr <- 1 to t) {
            val n = stdin.readLine.trim.toInt

            val result = utopianTree(n)

            printWriter.println(result)
        }

        printWriter.close()
    }
}
